aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -342,18 +343,49 @@ Tensor& eye_out_cpu(Tensor& result, int64_t n, int64_t m) {
 
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ full ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+namespace {
+
+// Performs dtype inference for full
+TensorOptions infer_full_options(
+  Scalar fill_value,
+  const TensorOptions& options) {
+
+  if (!options.has_dtype()) {
+    if (fill_value.isIntegral(true)) {
+      TORCH_WARN_ONCE(
+        "Deprecation warning: In PyTorch 1.6 torch.full with an integral ",
+        "fill_value will require a dtype or out argument. ",
+        "In PyTorch 1.7 an integral fill_value will return a LongTensor ",
+        "by default. ",
+        "Set the optional dtype or out arguments to suppress this warning. "
+      );
+    } else if (fill_value.isComplex()) {
+      auto scalar_type = (get_default_dtype() == ScalarType::Double) ?
```

Comment:
you can just use the get_default_complex_dtype once this PR: https://github.com/pytorch/pytorch/pull/34093/files#diff-2f560eadec29b69291ea551b2eea94a4R20 is landed